src/pyEQL/solution.py

```diff
@@ -59,118 +110,65 @@ def __init__(
-                    equilibrate() is called. Note that in this case, equilibrate() can distort the charge balance!
-            solvent: Formula of the solvent. Solvents other than water are not supported at this time.
-            engine: Electrolyte modeling engine to use. See documentation for details on the available engines.
-            database: path to a .json file (str or Path) or maggma Store instance that
-                contains serialized SoluteDocs. `None` (default) will use the built-in pyEQL database.
-            log_level: Log messages of this or higher severity will be printed to stdout. Defaults to 'ERROR', meaning
-                that ERROR and CRITICAL messages will be shown, while WARNING, INFO, and DEBUG messages are not. If set to None, nothing will be printed.
-            default_diffusion_coeff: Diffusion coefficient value in m^2/s to use in
-                calculations when there is no diffusion coefficient for a species in the database. This affects several
-                important property calculations including conductivity and transport number, which are related to the
-                weighted sums of diffusion coefficients of all species. Setting this argument to zero will exclude any
-                species that does not have a tabulated diffusion coefficient from these calculations, possibly resulting
-                in underestimation of the conductivity and/or inaccurate transport numbers.
-
-                Missing diffusion coefficients are especially likely in complex electrolytes containing, for example,
-                complexes or paired species such as NaSO4[-1]. In such cases, setting default_diffusion_coeff  to zero
-                is likely to result in the above errors.
-
-                By default, this argument is set to the diffusion coefficient of NaCl salt, 1.61x10^-9 m2/s.
+        # Initialize solution components (using a special dict)
+        self.components = FormulaDict({})
 
-        Examples:
-            >>> s1 = pyEQL.Solution({'Na+': '1 mol/L','Cl-': '1 mol/L'},temperature='20 degC',volume='500 mL')
-            >>> print(s1)
-            Components:
-            Volume: 0.500 l
-            Pressure: 1.000 atm
-            Temperature: 293.150 K
-            Components: ['H2O(aq)', 'H[+1]', 'OH[-1]', 'Na[+1]', 'Cl[-1]']
-        """
-        # create a logger and attach it to this class
-        self.log_level = log_level.upper()
+        # Process solutes only once
+        if solutes:
+            if isinstance(solutes, dict):
+                for formula, amount in solutes.items():
+                    self.add_solute(formula, amount)
+            elif isinstance(solutes, list) and all(isinstance(i, list) and len(i) == 2 for i in solutes):
+                for formula, amount in solutes:
+                    self.add_solute(formula, amount)
+            else:
```

Comment:
I think the _only_ change needed here in `solution.py` is to add `amount=interpret_units(amount)` right above each call to `add_solute`. Can you try that (keeping everything else in `solution.py` exactly as it is?